charts/core/src/Chart/hooks/useChart.ts

```diff
@@ -64,78 +146,39 @@ export function useChart({
     };
     window.addEventListener('resize', resizeHandler);
 
-    function onInitialRender() {
-      /**
-       * IMPORTANT NOTE: We use the presence of the handler to determine if the zoom click
-       * and drag is enabled or not. This is an exception! Typically we'd want a `zoomable` prop,
-       * or something similar to enable/disable, and a handler just to handle the action.
-       * We don't want to set a precedent with the pattern of implying functionality (like
-       * the click and drag action) based on the presence of a handler like this.
-       * This was deemed an exception to the rule due to the fact that the data is always
-       * controlled and there never exists a scenario where one would be useful without the other.
-       */
-      if (onZoomSelect) {
-        /**
-         * Zooming is built into echart via the toolbar. By default, a user
-         * has to click the "dataZoom" button to enable zooming. We however hide
-         * this button and want it turned on by default. This is done by dispatching
-         * an action to enable the "dataZoomSelect" feature, as if it were clicked.
-         */
-        chartInstance.dispatchAction({
-          type: 'takeGlobalCursor',
-          key: 'dataZoomSelect',
-          dataZoomSelectActive: true,
-        });
-
-        chartInstance.on('dataZoom', (params: any) => {
```

Comment:
This callback was moved into a function `handleDataZoom` outside of the `useEffect` for readability.